src/components/LoanCards/KvClassicLoanCardContainer.vue

```diff
@@ -0,0 +1,301 @@
+
+			// Some pages initially show loading cards without loan IDs
+			if (result.data?.lend?.loan) {
+				this.loan = result.data.lend.loan;
+				this.isBookmarked = result.data.lend.loan?.userProperties?.favorited ?? false;
+				this.userBalance = result.data?.my?.userAccount?.balance;
+			}
+
+			this.basketItems = result.data?.shop?.basket?.items?.values || null;
+		},
+		addToBasket(lendAmount) {
+			this.isAdding = true;
+			return setLendAmount({
+				amount: lendAmount,
+				apollo: this.apollo,
+				loanId: this.loanId,
+			}).then(() => {
+				this.isAdding = false;
+				this.$emit('add-to-basket', { loanId: this.loanId, success: true });
+				this.$kvTrackEvent(
+					'loan-card',
+					'add-to-basket',
+					null,
+					this.loanId,
+					this.lessThan25 ? this.amountLeft : 25
+				);
+			}).catch(e => {
+				this.$emit('add-to-basket', { loanId: this.loanId, success: false });
+				const msg = e?.[0]?.extensions?.code === 'reached_anonymous_basket_limit'
+					? e?.[0]?.message
+					: 'There was a problem adding the loan to your basket';
+				this.$kvTrackEvent('Lending', 'Add-to-Basket', 'Failed to add loan. Please try again.');
+				Sentry.captureException(e);
+				// Handle errors from adding to basket
+				if (hasBasketExpired(e?.[0]?.extensions?.code)) {
+					// eslint-disable-next-line max-len
+					this.$showTipMsg('There was a problem adding the loan to your basket, refreshing the page to try again.', 'error');
+					return handleInvalidBasket({
+						cookieStore: this.cookieStore,
+						loan: {
+							id: this.loanId,
+							price: lendAmount
+						}
+					});
+				}
+				this.$showTipMsg(msg, 'error');
+				this.isAdding = false;
+			});
+		},
+		toggleBookmark() {
+			if (!this.loanId) return;
+
+			// Set bookmark optimistically
+			this.isBookmarked = !this.isBookmarked;
+
+			bookmarkLoan(this.loanId, this.isBookmarked)
+				.then(() => {
+					this.$kvTrackEvent(
+						'Lending',
+						'Loan Favorite Toggled',
+						this.isBookmarked ? 'Favorite Loan Added' : 'Loan Favorite Removed',
+						this.isBookmarked,
+					);
+				}).catch(error => {
+					logFormatter(error, 'error');
+
+					this.$kvTrackEvent('Lending', 'Loan Favorite Toggled', 'Failed to toggle favorite.');
+
+					// Error occurred, change bookmark back due to failure
+					this.isBookmarked = !this.isBookmarked;
+				});
+		},
+		getCookie(name) {
+			return this.cookieStore.get(name);
+		},
+		setCookie(name, value, options) {
+			return this.cookieStore.set(name, value, options);
+		}
+	},
+	mounted() {
+		if (this.loan) {
+			// Already have a loan, so only setup watch query to handle changes in data
+			this.loadData();
+		} else {
+			// Don't have a loan yet, so setup viewport observer to prepare async loading
+			this.createViewportObserver();
+		}
+
+		// get route object
+		this.currentRoute = this.$route;
+	},
+	beforeDestroy() {
+		this.queryObserver.subscription.unsubscribe();
```

Comment:
This can be removed for now. I added a ticket to next sprint to address the subscriptions, since it's an existing issue in `ui`.